components/list/list-item-button-mixin.js

```diff
@@ -4,13 +4,32 @@ import { getUniqueId } from '../../helpers/uniqueId.js';
 import { ListItemMixin } from './list-item-mixin.js';
 
 export const ListItemButtonMixin = superclass => class extends ListItemMixin(superclass) {
+	static get properties() {
+		return {
+			/**
+			 * Disables the primary action button
+			 * @type {boolean}
+			 */
+			buttonDisabled : { type: Boolean, attribute: 'button-disabled' }
+		};
+	}
 
 	static get styles() {
 
 		const styles = [ css`
 			:host {
 				--d2l-list-item-content-text-color: var(--d2l-color-celestine);
 			}
+			:host([button-disabled][_hovering-primary-action]) .d2l-list-item-content,
+			:host([_focusing-primary-action]) .d2l-list-item-content {
+				--d2l-list-item-content-text-decoration: none;
+			}
+			:host([button-disabled]) .d2l-list-item-content {
```

Comment:
@oelhanafey Actually, I am not sure about the greying out. Can you confirm?

Reply:
I don't think we want to grey it out. A list-item-button with the button disabled should just appear like a regular list-item.

I think we also want to unset the color on the button since the current celestine color makes it look like it is clickable.